managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -375,6 +398,13 @@ protected OpAddEntry newObject(Recycler.Handle<OpAddEntry> recyclerHandle) {
         }
     };
 
+    public void recycleAfterClosed() {
+        if (STATE_UPDATER.get(this) != State.CLOSED) {
+            recycle();
+        }
+        closeFuture.thenAccept(ignore -> recycle());
+    }
```

Comment:
I think that there might be a better way to address this using reference counting. Another alternative is to skip recycling in cases where there's a potential issue. Recycling is not mandatory.

Reply:
> Another alternative is to skip recycling in cases where there's a potential issue. Recycling is not mandatory.

Good idea, improved.